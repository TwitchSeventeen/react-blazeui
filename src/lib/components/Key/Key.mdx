---
name: Key
---

import { Playground, PropsTable } from 'docz'
import Key from './index';

# Key

This styles a keyboard key. It can be used inline (ie it's not a block level element).

## Basic usage
<Playground>
    <Key>
        key
    </Key>
</Playground>

## Properties
<PropsTable of={Key} />