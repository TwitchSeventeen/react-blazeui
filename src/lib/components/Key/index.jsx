import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Key(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('u-kbd', className);
  return (
    <kbd
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </kbd>
  );
}

Key.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Key.defaultProps = {
  children: null,
  className: null,
  dataName: 'key',
};

export default Key;
