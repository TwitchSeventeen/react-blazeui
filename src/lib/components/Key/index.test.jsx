import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Key from './index';

describe('Key', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Key).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Key>
        key
      </Key>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
