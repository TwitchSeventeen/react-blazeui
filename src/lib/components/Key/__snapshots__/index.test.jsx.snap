// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key renders correctly 1`] = `
<DocumentFragment>
  <kbd
    class="u-kbd"
    data-name="key"
  >
    key
  </kbd>
</DocumentFragment>
`;
